Routes

Sisi-tech opened the pull request:
**User and Itinerary routes have been functionally tested using Postman. Below are the endpoints and their respective functionalities:**

**User Routes:**
```
Register (POST): http://localhost:8000/api/v1/user/register
Login (POST): http://localhost:8000/api/v1/user/login
```

**Itinerary Routes:**
```
Get All Itineraries (GET): http://localhost:8000/api/v1/itinerary/
Get Single Itinerary (GET): http://localhost:8000/api/v1/itinerary/:<id>
Create Itinerary (POST): http://localhost:8000/api/v1/itinerary/
Update Itinerary (PATCH): http://localhost:8000/api/v1/itinerary/:<id>
Delete Itinerary (DELETE): http://localhost:8000/api/v1/itinerary/:<id>
```

Sisi-tech committed:
connected to mongo db

Sisi-tech committed:
add lowercase: true, unique: true to emails, add timestamps: true to user schema. create user.js inside of controllers to handle register and login, create errors and middleware to handle errors and bad requests, add user.js inside of routes to handle register and login, and update app.js with userRouter.

Sisi-tech committed:
add zipCode to user schema

Sisi-tech committed:
Add itineraryItem controller to handle CRUD operations: getAllItineraryItems, getSingleItineraryItem, createItineraryItem, updateItineraryItem, deleteItineraryItem

Sisi-tech committed:
Refactored and tested itineraryItem routes; all endpoints verified functional with Postman

Sisi-tech committed:
User and Itinerary Routes

AmandaHockmuth reviewed (approved):
These are beautiful Sisi!
Do we want to keep a running list of the routes in the README.md file for front end reference? I still need to pop in and fix the ticketmaster endpoint I have in there, I did not enter it right before.

StuvaScript committed:
Initial commit

StuvaScript committed:
Edit some things to work with frontend

Commented out line 83 in modals/User.js file
according to an error in the console stating that
Mongoose.Schema implicitly creates a new email
index based on line 29 stating "unique: true"

Capitalize User.js file in modals folder.

Capitalize ItineraryItem route in itineraryItem.js

StuvaScript committed:
Changed named export to default export

In errors/unauthenticated.js file, changed the
export to a default export since we imported
it as a default import in middleware/
authentication.js

StuvaScript committed:
Edited controllers/itineraryItem.js file

Changed line 19 from 'req.user.userId' to
'req.user' to grab the correct info.

Changed line 26 from 'createdBy: userId' to
'user: userId' to match the model's schema.

StuvaScript committed:
Add code to return user ID

StuvaScript committed:
Changed invalid password error message

StuvaScript committed:
Changed invalid password message

Sisi-tech committed:
correct error message when email address not found.

jgoldenberg29 reviewed (approved):
Everything Looks great. The error handling is really well organized! I like it. 

The only thing I would change is for incorrect password, still just send back "invalid credentials" - we don't really want to tell folks which part of the login is incorrect just for a little extra security.

StuvaScript committed:
Merge pull request #33 from Code-the-Dream-School/Itinerary_edits_Stu_12.17.24

Itinerary edits stu 12.17.24

Sisi-tech committed:
Merge branch 'Routes' into Routes_edits_by_Stu_12.16.24

Sisi-tech committed:
Merge pull request #29 from Code-the-Dream-School/Routes_edits_by_Stu_12.16.24

Routes edits by stu 12.16.24

Sisi-tech committed:
Merge branch 'main' into Routes
